OracleLinux/7/README.md

```diff
@@ -1,36 +1,89 @@
-2. Install [Vagrant](https://vagrantup.com/)
+1. Install [Vagrant](https://vagrantup.com/)
 
 ## Getting started
+
 1. Clone this repository `git clone https://github.com/oracle/vagrant-boxes`
-2. cd vagrant-boxes/OracleLinux/7
-3. Run `vagrant status` to check Vagrantfile status and possible plugin(s) required
-4. Run `vagrant up`
+1. cd vagrant-boxes/OracleLinux/7
+1. Run `vagrant status` to check Vagrantfile status and possible plugin(s) required
+1. Run `vagrant up`
    1. The first time you run this it will provision everything and may take a while. Ensure you have a good internet connection!
-   2. The Vagrant file allows for customization.
-5. SSH into the VM either by using `vagrant ssh`
+   1. The Vagrant file allows for customization.
+1. SSH into the VM either by using `vagrant ssh`
    If required, by Vagrantfile you can also setup ssh port forwarding.
-6. You can shut down the box via the usual `vagrant halt` and the start it up again via `vagrant up`.
+1. You can shut down the box via the usual `vagrant halt` and the start it up again via `vagrant up`.
 
 ## Optional plugins
 
 When installed, this Vagrantfile will make use of the following third party Vagrant plugins:
+
+- [vagrant-env](https://github.com/gosuri/vagrant-env): loads environment
+variables from .env files;
 - [vagrant-proxyconf](https://github.com/tmatilai/vagrant-proxyconf): set
 proxies in the guest VMs if you need to access the Internet through proxy. See
 plugin documentation for the configuration.
 - [vagrant-reload](https://github.com/aidanns/vagrant-reload): reload the VM
 during provisioning to activate the latest kernel.
 
-To intall Vagrant plugins run:
-```
+To install Vagrant plugins run:
+
+```shell
 vagrant plugin install <name>...
 ```
 
+## Extending the project
+
+This project can easily be extended by running additional scripts during provisioning and optionally expose guest ports on the host.
+
+Environment variables are used to pass the parameters to the Vagrantfile:
+
+- `EXTEND`: coma separated list of extensions.  
+   For each extension a script `<extension>.sh` is run. Scripts must be located in the `scripts` or `scripts.local` directory.
+- `EXPOSE`: coma separated list of ports to expose in the format: `<host port>:<guest port>`
+
+Example: to extend the project with the `lamp` extension and expose the guest port 80 to 8080 on the host:
+
+```shell
+EXTEND=lamp EXPOSE=8080:80 vagrant up
+````
+
+Alternatively, if the `vagrant-env` plugin is installed variables can be defined in the `.env` or `.env.local` files.
+
+## Sample extensions
+
+### LAMP
+
+Provisions an Oracle Linux LAMP solution.
+
+Usage:
+
+```shell
+EXTEND=lamp
+EXPOSE=8080:80
+```
+
+Guest port "80" (Apache) is redirected to Host port "8080".  
+Once ready, you can test it by opening following URL on your Host OS: http://localhost:8080/info.php
+
+### Docker engine
+
+Installs and configures Docker engine with Btrfs as storage.
```

Comment:
Installs and configures Oracle Container Runtime for Docker. We do not ship the Docker CE and Docker Engine is not actually a product name. I'd also make this a link to the documentation: https://docs.oracle.com/en/operating-systems/oracle-linux/docker/

With that being said, it'd would probably be better to change `docker-engine` to `container-runtime` to avoid potentialy trademark usage issues (Docker is a registered trademark).